lib/routes/db.js

```diff
@@ -34,6 +34,15 @@ module.exports = function (app) {
 
   // Delete a database
   app.delete('/:db', function (req, res, next) {
+    if (req.query.rev) {
+      return utils.sendJSON(res, 400, {
+        error: 'bad_request',
+        reason: (
+          "You tried to DELETE a database with a ?rev= parameter. " +
+          "Did you mean to DELETE a document instead?"
+        )
+      });
+    }
```

Comment:
@janl has [a blog post](http://blog.couchdb.org/2014/04/08/the-little-things1-do-not-delete/) about this feature. Neat to see it implemented here. :)